howso/utilities/utilities.py

```diff
@@ -1393,3 +1393,37 @@ def format_confusion_matrix(confusion_matrix: dict[str, dict[str, int]]) -> tupl
             confusion_matrix_array[i, j] = confusion_matrix.get(actual_label, {}).get(predicted_label, 0)
 
     return confusion_matrix_array, row_labels
+
+
+def split_dataframe_into_chunks(df: pd.DataFrame, num_chunks: int) -> t.Generator[pd.DataFrame, None, None]:
+    """
+    Splits and yields a DataFrame in chunks using iloc. Np.array_split is deprecated.
+
+    Parameters
+    ----------
+    df : DataFrame
+        Pandas DataFrame to be split.
+    num_chunks : int
+        The number of chunks to split the df into.
+
+    Yields
+    ------
+    DataFrame
+        A DataFrame chunk.
+    """
+    total_rows = len(df)
+
+    if num_chunks > total_rows:
+        warnings.warn(
+            f"Number of chunks requested: {num_chunks} is greater than "
+            f"the number of rows in the DataFrame. Yielding the original DataFrame."
+        )
+        yield df
+
+    rows_per_chunk = total_rows // num_chunks
```

Comment:
This will yield num_chunks + 1 in many cases. See: `print(101 // 5)` => 20, so, you'd end up with a 6th chunk with a single row.

I suggest: `rows_per_chunk = math.ceil(total_rows / num_chunks)`